parser.py

```diff
@@ -194,32 +194,62 @@ def download_file_if_larger(url, filename, index, count, sleep_time):
        Returns whether the file is now known to be the largest available, and the number of bytes downloaded.
     """
     requests = import_module('requests')
+    imagesize = import_module('imagesize')
+
+    pref = f'{index:3d}/{count:3d} {filename}: '
     # Sleep briefly, in an attempt to minimize the possibility of trigging some auto-cutoff mechanism
     if index > 1:
-        print(f'{index}/{count}: Sleeping...', end='\r')
+        print(f'{pref}Sleeping...', end='\r')
         time.sleep(sleep_time)
     # Request the URL (in stream mode so that we can conditionally abort depending on the headers)
-    print(f'{index}/{count}: Requesting headers for {url}...', end='\r')
-    size_before = os.path.getsize(filename)
+    print(f'{pref}Requesting headers for {url}...', end='\r')
+    byte_size_before = os.path.getsize(filename)
     try:
         with requests.get(url, stream=True) as res:
             if not res.status_code == 200:
                 raise Exception('Download failed')
-            size_after = int(res.headers['content-length'])
-            if size_after > size_before:
+            byte_size_after = int(res.headers['content-length'])
+            if (byte_size_after != byte_size_before):
                 # Proceed with the full download
-                print(f'{index}/{count}: Downloading {url}...            ', end='\r')
+                print(f'{pref}Downloading {url}...', end='\r')
```

Comment:
The extra spaces were because otherwise with `end=\r` on the previous print the shorter message wouldn't overwrite all of the previous message on this line.

Unrelated question: I wonder if the streaming is working as I thought. The 'Downloading' mesage is never shown for very long.

Reply:
Ah, I was wondering about this. I did not really understand how `print` and `logging.info` interacted (and still don't).

I also noticed a similar problem when a URL was very long, so that the line with `Downloading {url}...` was longer than the log line that came after it, so that a part of the URL was still visible like in line ` 37/ 79`:

```
 35/ 79 media/866363488830095361-DAXwkwOXgAEQCY4.jpg: SKIPPED. Online version is same byte size, assuming same content. Not downloaded.
 36/ 79 media/866363488830095361-DAXwmLZWAAA9jea.jpg: SKIPPED. Online version is same byte size, assuming same content. Not downloaded.
 37/ 79 media/864232876539351040-uTWbzDlB_rr2kkh0.mp4: SKIPPED. Online version is same byte size, assuming same content. Not downloaded./vid/534x360/uTWbzDlB_rr2kkh0.mp4...
 38/ 79 media/855595638846894080-C98CmmCW0AAIePq.jpg: SKIPPED. Online version is same byte size, assuming same content. Not downloaded.
 39/ 79 media/855046691480887296-C928ZraXcAA8c5b.jpg: SKIPPED. Online version is same byte size, assuming same content. Not downloaded.
```

So I can revert the missing space characters, but this will not fix the whole problem. Maybe there should be a separate issue for this?

> Unrelated question: I wonder if the streaming is working as I thought. The 'Downloading' message is never shown for very long.

I'm testing this on a 1 Gbit/s connection, so these small downloads are always faster than I can read :D